vendor/brave-ios/Ledger/BATBraveLedger.mm

```diff
@@ -107,7 +107,9 @@ - (instancetype)initWithStateStoragePath:(NSString *)path
 
     ledgerClient = new NativeLedgerClient(self);
     ledger = ledger::Ledger::CreateInstance(ledgerClient);
-    ledger->Initialize();
+    ledger->Initialize(^(){
+      // TODO(jkuki) handle if needed
```

Comment:
Is there anything specific thats expected to be done _after_ initialize has completed? Or maybe even should we move certain calls to wait until after initialization, such as requesting balance, wallet props, etc.

Reply:
based on:
https://github.com/brave/brave-core/blob/caa3e591c9edd01a1b9321f877ac178ded3860e7/vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc#L89-L93

https://github.com/brave/brave-core/blob/caa3e591c9edd01a1b9321f877ac178ded3860e7/vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc#L244-L246

Should 

https://github.com/brave/brave-core/blob/caa3e591c9edd01a1b9321f877ac178ded3860e7/components/brave_rewards/browser/rewards_service_impl.cc#L962-L969

be passed as callback here?

https://github.com/brave/brave-core/blob/caa3e591c9edd01a1b9321f877ac178ded3860e7/components/brave_rewards/browser/rewards_service_impl.cc#L484

Reply:
Hmm I'm still a bit confused. I guess what you're saying though is that Initialize already does a bunch of stuff when it finishes internally, right? 

But I'm seeing that even in rewards_service_impl you binded it to `OnInitialize` that does nothing yet. Is there anything that we suspect would go in `OnInitialize`?

Reply:
My intentions would be to use `LoadLedgerState` instead of `OnInitialize` as it seems like this behaves as the callback to `Initialize`

Reply:
Hmm so in reality what it means to add a true callback to `Initialize`, actually means adding a callback to `LoadLedgerState` and passing the callback in there so that when LoadLedgerState finishes it calls the callback instead of `OnLedgerStateLoaded`

Reply:
Discussed with @NejcZdovc and he had the same conclusion. Will break out into separate PR to handle first